workflow/controller/operator_metrics_test.go

```diff
@@ -50,6 +50,8 @@ func TestBasicMetric(t *testing.T) {
 	_, err := wfcset.Create(wf)
 	assert.NoError(t, err)
 	woc := newWorkflowOperationCtx(wf, controller)
+	err = woc.setWorkflowSpecAndEntrypoint()
```

Comment:
If you have to add this function anytime a `WorkflowOperationCtx` then it's probably implemented incorrectly. Setting the `WorkflowSpec` and `Entrypoint` is an implementation detail of `WorkflowOperationCtx` and should not be exposed externally.

Is there a reason as to why you do this? If not, please refactor so that making these calls externally is not necessary.

Reply:
In Test, Controller function processNextItem will not be called. That's why we need to make call for `operate()` and this function on all test functions to mimic the real call flow.

Reply:
moved `loadWorkflowSpec` inside operate function